src/game_io/game_interactive_player_packet.cc

```diff
@@ -74,11 +74,7 @@ void GameInteractivePlayerPacket::read(FileSystem& fs, Game& game, MapObjectLoad
 			if (InteractiveBase* const ibase = game.get_ibase()) {
 				ibase->map_view()->scroll_to_map_pixel(center_map_pixel, MapView::Transition::Jump);
 
-				uint32_t const loaded_df =
-				   InteractiveBase::dfShowCensus | InteractiveBase::dfShowStatistics;
-				uint32_t const olddf = ibase->get_display_flags();
-				uint32_t const realdf = (olddf & ~loaded_df) | (display_flags & loaded_df);
-				ibase->set_display_flags(realdf);
+				ibase->set_display_flags(display_flags);
```

Comment:
Shouldn't the `dfDebug` flag get special treatment here?

Reply:
Good point. Saving a game in debug build and opening it in release shows the debug item.
I fixed it by ignoring the saved value and set it dependent on the build type.